pages/pools/[...poolData]/index.tsx

```diff
@@ -0,0 +1,118 @@
+//@ts-nocheck
+import Head from 'next/head';
+import styled from 'styled-components';
+import { ethers, BigNumber } from 'ethers';
+import { FC, useMemo, useCallback, useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import ViewPool from 'sections/Pools/ViewPool';
+
+import useGetPoolByIdQuery from 'queries/pools/useGetPoolByIdQuery';
+import { parsePool } from 'queries/pools/useGetPoolsQuery';
+import Connector from 'containers/Connector';
+import BaseModal from 'components/BaseModal';
+import Button from 'components/Button';
+import { nameToIdMapping } from 'constants/networks';
+
+const Pool: FC = () => {
+	const router = useRouter();
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+	const { network, provider } = Connector.useContainer();
+	const { poolData } = router.query;
+
+	const address = poolData && poolData.length > 0 ? poolData[0] : null;
+	const poolNetwork =
+		poolData && poolData.length > 1 && Object.keys(nameToIdMapping).indexOf(poolData[1]) !== -1
+			? poolData[1]
+			: null;
+
+	const poolAddress = (address ?? '') as string;
+
+	const poolNetworkId = nameToIdMapping[poolNetwork];
+
+	const poolQuery = useGetPoolByIdQuery({
+		id: poolAddress,
+		networkId: network.id,
+	});
+
+	const pool = useMemo(
+		// @ts-ignore
+		() => ((poolQuery?.data ?? null) != null ? parsePool(poolQuery.data) : null),
+		[poolQuery?.data]
+	);
+
+	const handleSwitchChain = useCallback(async () => {
```

Comment:
Code replication :( 

We can probably put this function in a file and import it